swarm/pss/pss.go

```diff
@@ -365,17 +371,25 @@ func (self *Pss) process(pssmsg *PssMsg) bool {
 			self.outbox <- pssmsg
```

Comment:
I think it is quite error-prone to create a go-routine for adding messages to the outbox channel. We should have some other mechanism to signal to the user if the outbox size is reached, and we should do something about it, not create unlimited number of go-routines that are blocked in memory.

I would just remove this goroutine altogether as a start.

Reply:
I originally was using arrays and counters which was slower according to benchmarks, but which had error on capacity issues. @zelig suggested use channels instead.

So you suggest to introduce a counter?

Reply:
using channels for the `outbox` is good, but when you have a buffered channel you should address what happens when the limit of the buffer is reached, and address this pressure. AFAIK you cannot have an unlimited buffer, as this basically means unlimited memory and this doesn't exist.

what you are doing now (goroutine which blocks once you reach the size of the outbox), means that you essentially have an unlimited number of goroutines that might block and you have a memory problem. we should not have that, but rather return an error and block if the outbox size is reached.